core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -777,6 +787,73 @@ NavigableMap<Long, InFlightBatch> cachedState() {
         return new ConcurrentSkipListMap<>(cachedState);
     }
 
+    // TODO: maxDeliveryCount should be utilized here once it is implemented
+    /**
+     * Apply acquisition lock to acquired records.
+     * @param memberId The member id of the client that is putting the acquisition lock.
+     * @param baseOffset The base offset of the acquired records.
+     * @param lastOffset The last offset of the acquired records.
+     * @param cachedStateEntry The cached state map entry for the acquired records which contains the entire batch
+     *                         from base offset to last offset.
+     */
+    public void applyAcquisitionLock(String memberId, long baseOffset, long lastOffset, Map.Entry<Long, InFlightBatch> cachedStateEntry) {
+        timer.add(new TimerTask(recordLockDurationMs) {
+            @Override
+            public void run() {
+                lock.writeLock().lock();
+                try {
+                    InFlightBatch inFlightBatch = cachedStateEntry.getValue();
+                    long localNextFetchOffset = nextFetchOffset;
+                    if (inFlightBatch.offsetState == null) {
+                        if (inFlightBatch.batchState() == RecordState.ACQUIRED) {
```

Comment:
I suggest a comment here to match the "case where batch has a valid offset state map." below.